File: boaty-app/src/app/services/ticket.service.ts
import { TicketModel } from 'src/app/model/ticket-model';
import { environment } from './../../environments/environment';
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class TicketService {

  constructor(private http: HttpClient) { }

  public getAllTickets() {
    return this.http.get<TicketModel[]>(`${environment.baseUrl}/tickets`);
  }

  public addTicket(ticket: TicketModel) {
    return this.http.post<TicketModel>(`${environment.baseUrl}/tickets`, ticket);
  }

  public deleteTicket(ticket: TicketModel){
    return this.http.delete<TicketModel>(`${environment.baseUrl}/tickets/${ticket.id}`);
  }
}

File: boaty-service/src/main/java/org/fmi/spring/boatyservice/service/TicketService.java
package org.fmi.spring.boatyservice.service;

import org.fmi.spring.boatyservice.model.Ticket;

import java.util.List;

public interface TicketService {

   List<Ticket> getAllTickets();

   Ticket addTicket(Ticket t);

   Ticket updateTicket(Ticket t);
}
